backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/VmHandler.java

```diff
@@ -1435,10 +1435,17 @@ public void autoSelectDefaultDisplayType(Guid srcEntityId,
 
         if (defaultDisplayType == null) {
             if (!displayGraphicsSupport.isEmpty()) {// when not found otherwise, let's take osinfo's record as the default
-                Map.Entry<DisplayType, Set<GraphicsType>> entry = displayGraphicsSupport.entrySet().iterator().next();
-                defaultDisplayType = entry.getKey();
+                // prioritize first display type based on osinfo
+                defaultDisplayType = displayGraphicsSupport.entrySet().iterator().next().getKey();
```

Comment:
this becomes a pattern of taking the first element and then iterate the collection and check the same first element again (also appeared in one of cpu related patches) - no reason not to iterate the first element differently within the loop or to continue looping with the iterator. for next time,,

Reply:
yes, i started with it. 
but then it will be something like:
```
if (defaultDisplayType == null) {
    defaultDisplayType = entry.getKey();
}
...
```
i liked this version better. but if you wish i can switch it.

Reply:
> ```
> if (defaultDisplayType == null) {
>     defaultDisplayType = entry.getKey();
> }
> ...
> ```
> 
> i liked this version better. but if you wish i can switch it.

it's a bit more expressive this way - your code should tell a story, so think about the story of the current code : "take the first element and then iterate all the elements and take any one that is not QXL" while the story here is "take the first one and override it with one that is not QXL". I find the second story to be a bit more clear (but you'll need to add `continue` there)